src/main/java/uk/gov/companieshouse/model/IllnessReason.java

```diff
@@ -1,27 +1,31 @@
 package uk.gov.companieshouse.model;
 
+import java.util.Collections;
 import java.util.List;
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
 
-public class IllnessReason{
+@Valid
+public class IllnessReason {
 
-    private final String illPerson;
-    private final String otherPerson;
-    private final String illnessStart;
-    private final boolean continuedIllness;
-    private final String illnessEnd;
-    private final String illnessImpactFurtherInformation;
-    private final List<Attachment> attachments;
+    @NotBlank(message = "illPerson must not be blank")
+    private String illPerson;
 
-    public IllnessReason(String illPerson, String otherPerson, String illnessStart, boolean continuedIllness,
-                     String illnessEnd, String illnessImpactFurtherInformation, List<Attachment> attachments) {
-        this.illPerson = illPerson;
-        this.otherPerson = otherPerson;
-        this.illnessStart = illnessStart;
-        this.continuedIllness = continuedIllness;
-        this.illnessEnd = illnessEnd;
-        this.illnessImpactFurtherInformation = illnessImpactFurtherInformation;
-        this.attachments = attachments;
-    }
+    @NotBlank(message = "illnessStart must not be blank")
+    private String illnessStart;
+
+    @NotNull(message = "continuedIllness must not be null")
+    private boolean continuedIllness;
+
+    @NotBlank(message = "illnessImpactFurtherInformation must not be blank")
+    private String illnessImpactFurtherInformation;
+
+    private String otherPerson;
+    private String illnessEnd;
+
+    @Valid
```

Comment:
I believe the set up here of checking a Valid List is still something that is needed as we have validation further on that checks the contents of an attachment and ensure that it's a valid attachment object.

Reply:
Gareth, I agree that checking the attachments are valid is a good thing. However if you follow the link my concern is that using a non bean breaks the recursive inspection.
Do you have a test or something that shows it is working as you expect rather than how I expect?

Reply:
Hi Jonathan. It looks like the @Valid annotation on list of object is correctly checking that the objects are correct. 
I have added two more tests to  check if is working as expected.

Reply:
Thanks Mo, I'm not sure why they aren't also throwing validation errors that there is no size or content type as those are notnull and notblank respectively. However I guess my concerns were old.